backend/src/converter/endpoints.ts

```diff
@@ -1,41 +1,48 @@
 import { validateConverter } from "./service";
-import { getCollections } from "../app";
+import { getCollections } from "../db";
 import { Request, Response } from "express";
 import { ObjectId } from "mongodb";
 
 export const getAllConverters = async (_req: Request, res: Response) => {
     const { converters } = getCollections();
-    try {
-        const result = await converters.find({}).toArray();
+    let result;
 
-        if (!result || result.length === 0) {
-            res.status(204).send();
-        } else {
-            res.status(200).json(result);
-        }
+    try {
+        result = await converters.find({}).toArray();
     } catch (err) {
-        res.status(500).json(err);
+        res.status(500).json({ error: "Database error" });
+        return;
+    }
+
+    if (!result || result.length === 0) {
+        res.status(404).json({ error: "No converters found" });
+    } else {
+        res.status(200).json(result);
     }
 };
 
 export const getConverterById = async (req: Request, res: Response) => {
     const { converters } = getCollections();
+    let id, result;
+
     try {
-        const id = new ObjectId(req.params.id);
+        id = new ObjectId(req.params.id);
+    } catch (err) {
+        res.status(400).json({ error: "Invalid id" });
+        return;
+    }
 
-        try {
-            const result = await converters.findOne({ _id: id });
+    try {
```

Comment:
Taki try catche też się powtarzają w wielu miejscach, tylko, że wykonywana jest inna akcja w `try`. Proponuję stworzyć jakąś osobną funkcję, która zawiera te stałe elementy jak zwracanie statusów w przypadku powodzenia w przypadku błędu, a to co się ma wykonać w try wrzucać jako parametr do tej funkcji (jako lambdę).

Reply:
Osobne zadanie na refaktoryzację